nominatim/tools/special_phrases.py

```diff
@@ -0,0 +1,245 @@
+    """
+    default_languages = [
+        'af', 'ar', 'br', 'ca', 'cs', 'de', 'en', 'es',
+        'et', 'eu', 'fa', 'fi', 'fr', 'gl', 'hr', 'hu',
+        'ia', 'is', 'it', 'ja', 'mk', 'nl', 'no', 'pl',
+        'ps', 'pt', 'ru', 'sk', 'sl', 'sv', 'uk', 'vi']
+    return config.LANGUAGES or default_languages
+
+
+def _get_wiki_content(lang):
+    """
+        Request and return the wiki page's content
+        corresponding to special phrases for a given lang.
+        Requested URL Example :
+            https://wiki.openstreetmap.org/wiki/Special:Export/Nominatim/Special_Phrases/EN
+    """
+    url = 'https://wiki.openstreetmap.org/wiki/Special:Export/Nominatim/Special_Phrases/' + lang.upper() # pylint: disable=line-too-long
+    return get_url(url)
+
+
+def _check_sanity(lang, phrase_class, phrase_type, pattern):
+    """
+        Check sanity of given inputs in case somebody added garbage in the wiki.
+        If a bad class/type is detected the system will exit with an error.
+    """
+    if len(pattern.findall(phrase_class)) < 1 or len(pattern.findall(phrase_type)) < 1:
+        LOG.error("Bad class/type for language %s: %s=%s", lang, phrase_class, phrase_type)
+
+
+def _process_amenity(db_connection, phrase_label, normalized_label,
+                     phrase_class, phrase_type, phrase_operator):
+    # pylint: disable-msg=too-many-arguments
+    """
+        Add phrase lookup and corresponding class and type to the word table based on the operator.
+    """
+    with db_connection.cursor() as db_cursor:
+        if phrase_operator == 'near':
+            db_cursor.execute("""SELECT getorcreate_amenityoperator(
+                              make_standard_name(%s), %s, %s, %s, 'near')""",
+                              (phrase_label, normalized_label, phrase_class, phrase_type))
+        elif phrase_operator == 'in':
+            db_cursor.execute("""SELECT getorcreate_amenityoperator(
+                              make_standard_name(%s), %s, %s, %s, 'in')""",
+                              (phrase_label, normalized_label, phrase_class, phrase_type))
+        else:
+            db_cursor.execute("""SELECT getorcreate_amenity(
+                              make_standard_name(%s), %s, %s, %s)""",
+                              (phrase_label, normalized_label, phrase_class, phrase_type))
+
+
+def _create_place_classtype_table_and_indexes(db_connection, config, class_type_pairs):
+    """
+        Create table place_classtype for each given pair.
+        Also create indexes on place_id and centroid.
+    """
+    LOG.warning('Create tables and indexes...')
+
+    sql_tablespace = config.TABLESPACE_AUX_DATA
+    if sql_tablespace:
+        sql_tablespace = ' TABLESPACE '+sql_tablespace
+
+    with db_connection.cursor() as db_cursor:
+        db_cursor.execute("CREATE INDEX idx_placex_classtype ON placex (class, type)")
+
+    for pair in class_type_pairs.items():
```

Comment:
I get a 

```
  File "Nominatim/nominatim/tools/special_phrases.py", line 159, in _create_place_classtype_table_and_indexes
    for pair in class_type_pairs.items():
AttributeError: 'set' object has no attribute 'items'
```

Reply:
This error is my bad as I was only testing the first part of the process... Fixed.